pkg/querier/batch/merge.go

```diff
@@ -70,6 +77,18 @@ func newMergeIterator(it iterator, cs []GenericChunk) *mergeIterator {
 	return c
 }
 
+func (c *mergeIterator) putPointerValuesToThePool(b chunk.Batch) {
+	if b.ValueType == chunkenc.ValHistogram || b.ValueType == chunkenc.ValFloatHistogram {
```

Comment:
nit: let's make two separate loops, no need to check the type for each item as one Batch can only hold one type